pyproject.toml

```diff
@@ -264,3 +277,23 @@ doctest_optionflags = [
     "IGNORE_EXCEPTION_DETAIL",
     "ELLIPSIS",
 ]
+
+[tool.conda-lock]
+channels = [
+    "conda-forge",
+    "defaults",
+]
+platforms = [
+    "osx-arm64",
+    "linux-64",
+]
+channel-priority = "strict"
+name = "pudl-dev"
+
+[tool.conda-lock.dependencies]
+recordlinkage = {source = "pypi"}
+sphinx-autoapi = {source = "pypi"}
+pip = ">=23,<24"
+python = ">=3.11,<3.12"
+python-snappy = ">=0.6,<1"
+sqlite = ">=3.36,<4"
```

Comment:
Mostly I left the `conda-lock` config here for discussion.  See the `conda-lock.yml` file as well (all 15000 lines of it!)

With this `pyproject.toml` all of our dependencies except for `recordlinkage` and `sphinx-autoapi` are installed from `conda-forge` (and those two packages alone are singled out for installation from PyPI).

Our programmatic non-python dependencies (snappy, sqlite..) also get installed from `conda-forge`.

We could add `jupyterlab` and the other interactive dependencies from `devtools/environment.yml` into the `dev` optional dependencies above, and `pandoc` / `nodejs` here if we wanted, and eliminate `devtools/environment.yml` and probably our other `*-environment.yml` files.

Would need to add some infrastructure for updating the lockfile on a regular basis and get used to using it to install our environment, but it does seem to be working.